### PR TITLE
Documentation for named ports.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add documentation for the new named ports feature https://github.com/projectcalico/felix/pull/1534

## Todos

- [x] Tests N/A
- [x] Documentation
- [x] Release note

## Release Note

```release-note
Calico now implements support for named ports in policy rules and maps named ports through from Kubernetes Pod and NetworkPolicy resources.  This allows you to attach a name to a particular port on a particular endpoint and then match on the name rathe than the port number.  Different endpoints can use different port numbers for the same named port.
```

### CODE REVIEW

File: _includes/master/endpointport.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+An EndpointPort associates a name with a particular TCP/UDP port of the endpoint, allowing it to
+be referenced as a named port in [policy rules]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/policy#entityrule).
+
+| Field       | Description                      | Accepted Values   | Schema | Default    |
+|-------------|----------------------------------|-------------------|--------|------------|
+| name        | The name to attach to this port, alowing it to be referred to in [policy rules](./policy#entityrule).  Names must be unique within an endpoint.  | | string |            |
```
Comment (bcreane) [inline _includes/master/endpointport.md]: alowing -> allowing

File: _includes/master/endpointport.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+An EndpointPort associates a name with a particular TCP/UDP port of the endpoint, allowing it to
+be referenced as a named port in [policy rules]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/policy#entityrule).
```
Comment (neiljerram) [inline _includes/master/endpointport.md]: Does this link render correctly with a space (apparently) between the `]` and the `(` ?  It's a useful TIL if so!

File: _includes/master/endpointport.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+| Field       | Description                      | Accepted Values   | Schema | Default    |
+|-------------|----------------------------------|-------------------|--------|------------|
+| name        | The name to attach to this port, allowing it to be referred to in [policy rules](./policy#entityrule).  Names must be unique within an endpoint.  | | string |            |
```
Comment (neiljerram) [inline _includes/master/endpointport.md]: If this shorter link works, can we use that above as well?

File: _includes/master/endpointport.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+| protocol    | The protocol of this named port. | tcp, udp          | string |            |
+| port        | The workload port number.        | 1-65535           | int    |            |
+
+> ** NOTE **
+>
+> On their own, EndpointPort entries don't result in any change to the connectivity of the port.
+> They only have an effect if they are referred to in policy.
```
Comment (neiljerram) [inline _includes/master/endpointport.md]: Please adapt this according to current master, where the standard Note formatting has slightly changed.  (E.g. look for `alert-info` in `master/getting-started/bare-metal/bare-metal.md`.)

File: _includes/master/ports.md [hunk 5]
```diff
@@ -1,6 +1,30 @@
...
+port on port 80 and others on port 8080. In each workload, you could create a named port called 
+`http-port` that maps to the correct local port.  Then, in a rule, you could refer to the name 
+`http-port` instead of writing a different rule for each type of server.
+
+> **NOTE**
+> 
+> Since each named port may refer to many endpoints (and Calico has to expand a named port into
+> a set of endpoint/port combinations), using a named port is considerably more expensive in terms 
+> of CPU than using a simple numeric port.  We recommend that they are used sparingly, only where 
+> the extra indirection is required.
```
Comment (neiljerram) [inline _includes/master/ports.md]: Another Note to adapt to latest standard.

File: _includes/master/endpointport.md (hunk 2 above)
Comment (fasaxc) [reply _includes/master/endpointport.md]: No, there's no space, it's just how github renders it if the window is too small.

**Pull request merged**